practice-two/src/components/FormValidation/FormValidation.jsx

```diff
@@ -0,0 +1,183 @@
+import { memo, useCallback, useState } from 'react';
+import './FormValidation.css';
+import { InputValidate, Button } from '@components';
+import { BUTTON_VARIANTS } from '@constants';
+import { Validation } from '@helpers';
+
+const FormValidation = (props) => {
+  const { onShowForm } = props;
+
+  const [formData, setFormData] = useState({
+    name: '',
+    avatar: '',
+    mail: '',
+    phoneNumber: '',
+    description: '',
+    address: '',
+    gender: 'Male',
+  });
+
+  const [errors, setErrors] = useState({
+    name: '',
+    avatar: '',
+    mail: '',
+    phoneNumber: '',
+    description: '',
+    address: '',
+  });
+
+  // Set value for form data
+  const handleChange = useCallback(
+    (e) => {
+      setFormData((prevFormData) => ({
+        ...prevFormData,
+        [e.target.name]: e.target.value,
```

Comment:
cache the target event

Reply:
![image](https://github.com/datdragon777/React-Training/assets/74806826/caeff0c3-758f-4d07-b1fa-0afdcbb89f83)